src/components/Flowbite/index.tsx

```diff
@@ -0,0 +1,32 @@
+import { FC, PropsWithChildren, useLayoutEffect, useMemo } from 'react';
+import useDarkMode from '../../hooks/useDarkMode';
+import { ThemeContext } from '../../contexts/ThemeContext';
+
+export type FlowbiteProps = PropsWithChildren<{
+  children: any;
+  dark?: boolean;
+  usePreferences?: boolean;
+}>;
+
+export const Flowbite: FC<FlowbiteProps> = ({ children, dark, usePreferences = true }) => {
+  const [mode, setMode, toggleMode] = useDarkMode(usePreferences);
+
+  useLayoutEffect(() => {
+    if (dark) {
+      if (setMode != null) {
+        setMode('dark');
+      }
+      document.documentElement.classList.add('dark');
+    }
+  }, [dark, setMode]);
+
+  const value = useMemo(
```

Comment:
We can rename this to `themeContextValue`

Reply:
I'll change it, but I don't think that it will make any big difference to code readability. The variable scope context is very short. In these cases, short names are better.

Quick example, you don't need to say to a `createUser` function that an internal variable called `lastname` is `useLastName`, or `age` is `userAge`. :)